Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift

```diff
@@ -1233,100 +1233,131 @@ class MessageDTO_Tests: XCTestCase {
+            authorUserId: .unique,
+            createdAt: Date(timeIntervalSince1970: 1),
             channel: channel
         )
 
-        let deletedReplyInChannel: MessagePayload = .dummy(
-            type: .deleted,
-            messageId: deletedReplyInChannelId,
-            parentId: messageId,
-            showReplyInChannel: true,
+        XCTAssertTrue(try checkChannelMessagesPredicate(channelId: channelId, message: message))
+    }
+
+    func test_channelMessagesPredicate_shouldNotIncludeEphemeralMessagesOnThreads() throws {
+        let channelId: ChannelId = .unique
+        let channel = ChannelDetailPayload.dummy(cid: channelId)
+        let message: MessagePayload = .dummy(
+            type: .ephemeral,
+            messageId: .unique,
+            parentId: .unique,
             attachments: [],
-            authorUserId: authorId,
-            createdAt: Date(timeIntervalSince1970: 5),
+            authorUserId: .unique,
+            createdAt: Date(timeIntervalSince1970: 1),
             channel: channel
         )
 
-        let messages = [message, existingReplyNotInChannel, existingReplyInChannel, deletedReplyNotInChannel, deletedReplyInChannel]
-        try database.writeSynchronously { session in
-            try messages.forEach { messagePayload in
-                try session.saveMessage(payload: messagePayload, for: channelId, syncOwnReactions: true)
-            }
-        }
+        XCTAssertFalse(try checkChannelMessagesPredicate(channelId: channelId, message: message))
+    }
 
-        let request = NSFetchRequest<MessageDTO>(entityName: MessageDTO.entityName)
-        request.sortDescriptors = [NSSortDescriptor(keyPath: \MessageDTO.defaultSortingKey, ascending: true)]
-        request.predicate = MessageDTO.channelMessagesPredicate(
-            for: channelId.rawValue,
-            deletedMessagesVisibility: .visibleForCurrentUser,
-            shouldShowShadowedMessages: false
+    func test_channelMessagesPredicate_shouldIncludeRegularMessages() throws {
+        let channelId: ChannelId = .unique
+        let channel = ChannelDetailPayload.dummy(cid: channelId)
+        let message: MessagePayload = .dummy(
+            type: .regular,
+            messageId: .unique,
+            attachments: [],
+            authorUserId: .unique,
+            createdAt: Date(timeIntervalSince1970: 1),
+            channel: channel
         )
 
-        var retrievedMessages: [MessageDTO] = []
-        do {
-            retrievedMessages = try database.viewContext.fetch(request)
-        } catch {
-            XCTFail("Failed to retrieve messages")
-        }
-
-        XCTAssertEqual(retrievedMessages.count, 2)
-        XCTAssertEqual(
-            retrievedMessages.map(\.id),
-            [messageId, existingReplyInChannelId]
+        XCTAssertTrue(try checkChannelMessagesPredicate(channelId: channelId, message: message))
+    }
+    
+    func test_channelMessagesPredicate_shouldIncludeDeletedMessages() throws {
```

Comment:
Should we also add a test for thread reply being deleted with showReplyInChannel: true?